miden-lib/src/transaction/inputs.rs

```diff
@@ -320,6 +320,11 @@ fn add_input_notes_to_advice_inputs(
         // insert note authentication path nodes into the Merkle store
         match input_note {
             InputNote::Authenticated { note, proof } => {
+                let note_block_header = tx_inputs
+                    .block_chain()
+                    .get_block(proof.location().block_num())
+                    .unwrap_or(tx_inputs.block_header());
```

Comment:
I think here we are implicitly assuming that if the block header is not found in the chain MMR it must be the last bock header. I'm wondering, if we should make this assumption more explicit. For example, something like:
```rust
let note_block_header = proof.location().block_num() == tx_inputs.block_header().block_num() {
    tx_inputs.block_header()
} else {
   ...
}
```